invenio_users_resources/services/generators.py

```diff
@@ -108,3 +110,23 @@ def query_filter(self, identity=None, **kwargs):
                     return dsl.Q("term", id=need.value)
 
         return []
+
+
+class UserModeration(Generator):
+    """Allows user-management."""
+
+    def __init__(self):
+        """Constructor."""
+        super().__init__()
+
+    def needs(self, **kwargs):
+        """Enabling Needs."""
+        return [user_moderation_action]
+
+    def query_filter(self, identity=None, **kwargs):
+        """Filters for current identity as system process."""
+        if user_moderator in identity.provides:
+            # TODO we might want to have an umbrella "user_moderation" for some fields
```

Comment:
Ideally, a user moderator could see fields related to user moderation (e.g. `blocked_at`). We could add a `match` query for each field. E.g. `dsl.Q("match", **{field: "public"})`, but that means we need to add one query per field added. 

IMO that would not be easy to maintain / scale, therefore we could consider an umbrella for these fields (e.g. `user_moderation.blocked_at`).

For now, a `UserModerator` can see all fields from a user.